### PR TITLE
Fix mob sprite drift when doing anims while floating

### LINKED ISSUE #9119: Rubber ducky sprte raise bug.
**Problem Description**:
When flying in space and constantly hitting yourself with rubber ducky, your sprite moves slightly higher than your actual character's position.

**What did you expect to happen**: 
To squack myself to oblivion.

**What happened instead**:
Literally squacked myself to oblivion.

**Why is this bad/What are the consequences:**
Prolly gamebreaking a little.

**Steps to reproduce the problem**:
0. Use hand teleporter to get into open space. (Probably not the cause)
1. Fly to north/south in space, while constantly hitting yourself with captain's rubber ducky.
2. See your sprite raise up.
3. Panic.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: I was playing vox that time. If you hit anything, or do flip and fall down, sprite goes back to normal. Pretty much any sprite update, i guess. It also resets the moment you touch floor or plating.
image
Red circle is voxxy's actual position.

### LINKED ISSUE #13686: Melee Attacks in Space drift your Sprite Off-screen
**Issue Description**:

Doing multiple melee attacks from space will slowly drift your sprite up, until it goes off the edge of the screen.

**What did you expect to happen**: 

My sprite would stay in the same place.

**What happened instead**:

Instead it slowly drifted upward until it looked like it was on a different tile, and kept drifting further upward with each attack until it drifted off the top of the screen.

**Why is this bad/What are the consequences**:

**Steps to reproduce the problem**:

1. Get Paramedic hardsuit and Dsword.
2. Go into space.
3. Repeatedly attack windows from space over and over again.

**When did the problem start happening**:

**Extra information**:

Getting shocked or lying down resets the position of your sprite to where it's supposed to be.

This is a screenshot of me appearing to be 2 tiles above where I actually am.
image

### PR DESCRIPTION
**Pull request by dearmochi**
## What Does This PR Do

- Fixes most (if not all) cases of sprite drifting on floating mobs (no gravity/in space) that perform an animation (attacking, jitter). The fix works by making the floating animation (among others) use the `ANIMATION_RELATIVE` flag so that it doesn't use outdated values (fixes #9119, fixes #13686). We should be using that flag where possible anyway.
- Minor code clean-up, improve `mob/living/proc/float` readability

## Why It's Good For The Game

Prevents mob sprites from drifting away, providing a clear view on their actual location. That glitch may be used maliciously to harm people without them being able to do anything

## Changelog
:cl:
fix: Fix cases of sprite drifting while floating
/:cl:

### CODE REVIEW

File: code/__HELPERS/matrices.dm [hunk 1]
```diff
@@ -24,6 +24,9 @@
 		animate(transform = matrices[i], time = speed)
 		//doesn't have an object argument because this is "Stacking" with the animate call above
 		//3 billion% intentional
+	if(isliving(src))
```
Comment (Fox-McCloud) [inline code/__HELPERS/matrices.dm:27]: this is nowhere near acceptable and a pretty egregious violation of OOP.

Comment (dearmochi) [reply code/__HELPERS/matrices.dm:27]: Whoops, didn't even notice this was a helper - I'll fix that in a moment

**Comment by dearmochi** [comment]
Apologies for the ugly history, Travis threw a fit and I tried fixing it. First attempt didn't go so well but second one went better

**Comment by AffectedArc07** [comment]
Other PR merged

**Pull request closed without merging**